pkg/asset/installconfig/aws/validation.go

```diff
@@ -59,7 +60,13 @@ func Validate(ctx context.Context, meta *Metadata, config *types.InstallConfig)
 	}
 	return allErrs.ToAggregate()
 }
-
+func validateRegion(region string) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if _, ok := endpoints.AwsPartition().Regions()[region]; !ok {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("platform", "aws", "region"), region, "invalid region"))
```

Comment:
It is unnecessary (and likely less optimal execution) to call an append when there are not multiple errors to be combined and returned. Instead, return `field.ErrorList{field.Invalid(field.NewPath("platform", "aws", "region"), region, "invalid region")}` here and return `nil` below. You can refer to `func validateAMI()` below for similar usage.

Reply:
Is this approach correct? I was under the (vague) impression that we allowed custom regions along with custom service endpoints? There was a validation test like that here.
https://github.com/openshift/installer/blob/46acb726bbebb7d2511dc7f83ed92b6385eb286b/pkg/asset/installconfig/aws/validation_test.go#L445

Reply:
No it is not correct. I just realized that I never submitted my comment from yesterday. See below.